### DISCUSSION TITLE
Hooks http context aware

### DISCUSSION THREAD

**Discussion by gkielwasser** [directus/directus#8043] [Archive]
Hi,

I would like to see if there is a way of synchronizing the Directus  auth session with a session in another server.

Naively I thought to the webhooks: ```'auth.login.before', 'auth.logout.before', 'auth.refresh.before'``` where I would have done my request to the other server and pulled back my Bearer token. Then this token would have been attached to the response of the orginal request.
The point is a hook has no idea from which http context it comes from (if there is any).

Is there another mechanism that would allow me to do this appart from creating 3 new custom endpoints (login,logout,refresh) ?

Thanks for feeback

**Comment by rijkvanzanten** [comment]
When you say 

> I would like to see if there is a way of synchronizing the Directus auth session with a session in another server.


What do you mean exactly? Do you have any specific examples in mind?

**Reply by gkielwasser** [reply]
Hi,
Let's say I have one server with Directus and another one with its own authentication system. 
I wanted to use the Directus auth session hook to do a flow that would like this:

``` auth.login.before ``` attempt to authenticate the user in the other server. Attach the retrieved Bearer token to the directus login response and allow login. The front application would be able to read both authorization token.

``` auth.refresh.before ``` attempt to refresh the user session in the other server. Attach the retrieved Bearer token to the directus response and allow refresh.

``` auth.logout.before ``` attempt to logout the user in another server. Remove the Bearer token of the other server and allow logout.

**Reply by joselcvarela** [reply]
If your server support OAuth you can use it to login into Directus.
https://docs.directus.io/guides/api-config/#oauth-single-sign-on-sso-openid

You can check here if supports:
https://github.com/simov/grant

Otherwise I think you would need to do a custom login flow.

**Reply by rijkvanzanten** [reply]
@gkielwasser that flow sounds very similar to OpenID/LDAP/SAML. Does your external service support either one of those standards? When it comes to auth flows like that, I personally prefer relying on a known standard that can be configured rather than trying to reinvent the wheel
